### ISSUE TITLE
Proxy is not working when accessing https websites (your connection to this site is not secure)

### ISSUE THREAD

**Issue by smartdev10** [Cuadrix/puppeteer-page-proxy#9]
Hi , first of all thanks for this package  , i hope someone can help me with this 
my issue is 
connecting to my gmail account is not successful this the code i used , when i use puppeteer.launch
with --args=proxy it works , but with connect and usProxy it is not working

my goal is to connect to a list of gmail accounts each with it's own proxy

```
browser = await puppeteer.connect({
    browserWSEndpoint:"ws://127.0.0.1:58447/devtools/browser/1351f816-f88c-4954-a1f1-9c7e93d2f930",
  })

  
  context = await browser.createIncognitoBrowserContext();
  
  page = await context.newPage();


  await useProxy(page, proxy);
  await page.setViewport({ width: 0, height: 0 });
  // 3. Navigate to URL
  await page.goto('http://mail.google.com/mail/' ,  { waitUntil: 'networkidle2' });
```

this 400 error page after i perform automated login
![image](https://user-images.githubusercontent.com/57545929/76566841-e908a880-64ad-11ea-9bb6-320ea5ca8618.png)

the same error using 

```
 page.on('request', req => {
        useProxy(req, proxy); 
    });
```

**Comment by Sara2009** [comment]
I use a Proxy software like fiddler or whistle. It occurs an error net::ERR_FAILED. I cannot get the response. I wander how to solve this problem.

**Comment by Bouzazi** [comment]
I think it's due to it redirecting to http not https
Just opened an issue here #12

**Comment by eddylapis** [comment]
I have a problem when the page has PostMessage cross domain access。

Failed to execute 'postMessage' on 'DOMWindow': The target origin provided ('https://a.domain.com') does not match the recipient window's origin ('https://b.domain.com').


Net:: err_failed is returned when some XHR is submitted

**Comment by Cuadrix** [comment]
The 'net::ERR_FAILED' is fixed in v1.2.4
Download the latest package

Read about the https issue [here](https://github.com/Cuadrix/puppeteer-page-proxy#faq)
